### PR TITLE
Provide more complete DOCTYPE

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @steveklabnik

### CODE REVIEW

**Comment by eefriedman** [comment]
Why would we do this?  The HTML5 specification says that `<!DOCTYPE html>` is recommended.  Also, the specific doctype you're suggesting turns on quirks mode, which seems like a bad idea.

**Comment by GuillaumeGomez** [comment]
It's because `:hover` wasn't working on a div on firefox (which is weird since it works just fine on chrome). I checked on the internet and that's one of the solution I found. However, if you have better, I'll gladly hear it.

**Comment by eefriedman** [comment]
You'll have to be a bit more specific; `div:hover` works in every browser whether or not quirks mode is enabled.

File: src/librustdoc/html/layout.rs [hunk 1]
```diff
@@ -36,7 +36,8 @@ pub fn render<T: fmt::Display, S: fmt::Display>(
     -> io::Result<()>
 {
     write!(dst,
-r##"<!DOCTYPE html>
+r##"<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
+     "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```
Comment (steveklabnik) [inline src/librustdoc/html/layout.rs]: yeah, this changes from HTML5 to XHTML 1.0, which we shouldn't do. this is the correct doctype

**Pull request closed without merging**